examples/with-stripe-typescript/app/actions/stripe.ts

```diff
@@ -29,13 +34,24 @@ export async function createCheckoutSession(data: FormData): Promise<void> {
           },
         },
       ],
-      success_url: `${headers().get(
-        "origin",
-      )}/donate-with-checkout/result?session_id={CHECKOUT_SESSION_ID}`,
-      cancel_url: `${headers().get("origin")}/donate-with-checkout`,
-    });
+      ...(ui_mode === 'hosted' && {
+        success_url: `${headers().get(
+          'origin'
+        )}/donate-with-checkout/result?session_id={CHECKOUT_SESSION_ID}`,
+        cancel_url: `${headers().get('origin')}/donate-with-checkout`,
+      }),
+      ...(ui_mode === 'embedded' && {
+        return_url: `${headers().get(
+          'origin'
+        )}/donate-with-embedded-checkout/result?session_id={CHECKOUT_SESSION_ID}`,
+      }),
```

Comment:
Might be worth pulling out `headers().get('origin')` into a variable here. It's repeated lots now.